Description: Cast %u format specifiers to unsigned long to prevent compiler
 warnings on 32-bit and 64-bit platforms. 
Author: Hugh McMaster <hugh.mcmaster@outlook.com>
Last-Update: 2018-04-02

Index: libexif-0.6.21/libexif/canon/exif-mnote-data-canon.c
===================================================================
--- libexif-0.6.21.orig/libexif/canon/exif-mnote-data-canon.c
+++ libexif-0.6.21/libexif/canon/exif-mnote-data-canon.c
@@ -267,8 +267,8 @@ exif_mnote_data_canon_load (ExifMnoteDat
 			if ((dataofs + s < s) || (dataofs + s < dataofs) || (dataofs + s > buf_size)) {
 				exif_log (ne->log, EXIF_LOG_CODE_DEBUG,
 					"ExifMnoteCanon",
-					"Tag data past end of buffer (%u > %u)",
-					dataofs + s, buf_size);
+					"Tag data past end of buffer (%lu > %u)",
+					(unsigned long)(dataofs + s), buf_size);
 				continue;
 			}
 
Index: libexif-0.6.21/libexif/fuji/exif-mnote-data-fuji.c
===================================================================
--- libexif-0.6.21.orig/libexif/fuji/exif-mnote-data-fuji.c
+++ libexif-0.6.21/libexif/fuji/exif-mnote-data-fuji.c
@@ -225,7 +225,7 @@ exif_mnote_data_fuji_load (ExifMnoteData
 				(dataofs + s >= buf_size)) {
 				exif_log (en->log, EXIF_LOG_CODE_CORRUPT_DATA,
 						  "ExifMnoteDataFuji", "Tag data past end of "
-					  "buffer (%u >= %u)", dataofs + s, buf_size);
+					  "buffer (%lu >= %u)", (unsigned long)(dataofs + s), buf_size);
 				continue;
 			}
 
Index: libexif-0.6.21/libexif/olympus/exif-mnote-data-olympus.c
===================================================================
--- libexif-0.6.21.orig/libexif/olympus/exif-mnote-data-olympus.c
+++ libexif-0.6.21/libexif/olympus/exif-mnote-data-olympus.c
@@ -482,8 +482,8 @@ exif_mnote_data_olympus_load (ExifMnoteD
 			    (dataofs + s > buf_size)) {
 				exif_log (en->log, EXIF_LOG_CODE_DEBUG,
 					  "ExifMnoteOlympus",
-					  "Tag data past end of buffer (%u > %u)",
-					  dataofs + s, buf_size);
+					  "Tag data past end of buffer (%lu > %u)",
+					  (unsigned long)(dataofs + s), buf_size);
 				continue;
 			}
 
Index: libexif-0.6.21/libexif/pentax/exif-mnote-data-pentax.c
===================================================================
--- libexif-0.6.21.orig/libexif/pentax/exif-mnote-data-pentax.c
+++ libexif-0.6.21/libexif/pentax/exif-mnote-data-pentax.c
@@ -308,7 +308,7 @@ exif_mnote_data_pentax_load (ExifMnoteDa
 				(dataofs + s > buf_size)) {
 				exif_log (en->log, EXIF_LOG_CODE_DEBUG,
 						  "ExifMnoteDataPentax", "Tag data past end "
-					  "of buffer (%u > %u)", dataofs + s, buf_size);
+					  "of buffer (%lu > %u)", (unsigned long)(dataofs + s), buf_size);
 				continue;
 			}
 
